itest/assets_test.go

```diff
@@ -882,44 +896,61 @@ func payInvoiceWithAssets(t *testing.T, payer, rfqPeer *HarnessNode,
 		sendReq.MaxShardSizeMsat = 80_000_000
 	}
 
+	var rfqBytes []byte
+	manualRfq.WhenSome(func(i rfqmsg.ID) {
+		copy(rfqBytes, i[:])
```

Comment:
`copy()` is tricky if the target slice is nil or empty, nothing happens. So you need to do `rfqBytes = make([]byte, len(i[:]))` first.

Reply:
Nice catch! Interestingly enough the tests did not fail as the daemon would automatically acquire fresh quotes for the payment. Specifying an rfqID here was a effectively a noop.

The reason this wasn't caught by the next test case (which requires you use a specific rfq scid and none else) is because it is enforced by the receiver, while crafting the invoice.